### PR TITLE
[New Rule] RC Script Creation

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
The /etc/rc.local file is used to start custom applications, services, scripts or commands during start-up. The rc.local file has mostly been replaced by Systemd, however through the "systemd-rc-local-generator", rc.local files can be converted to services that run at boot. Adversaries may alter rc.local to execute malicious code at start-up, and gain persistence onto the system. 
### Detection
By looking at a sequence of rc.local creation and a chmod event in a small timespan, I expect to minimize false positives and detect both manual and automated persistence behavior. rc.local has to be executable in order for the systemd-rc-local-generator file to load it into the boot process and execute it. 

image

### CODE REVIEW

File: rules/linux/persistence_rc_script_creation.toml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+type = "eql"
+
+query = '''
+sequence by user.id, host.id with maxspan=15s
+[file where event.type == "creation" and
+file.path : "/etc/rc.local"]
+
+[process where event.type == "start" and
+process.name == "chmod" and
+process.args : ("+x", "/etc/rc.local")]
```
Comment (DefSecSentinel) [inline rules/linux/persistence_rc_script_creation.toml:68]:
```suggestion
sequence by user.id, host.id with maxspan=15s
[file where event.type == "creation" and
file.path : "/etc/rc.local"]
[process where event.type == "start" and
process.name == "chmod" and
process.args : "+x"
process.args : "/etc/rc.local"]
```

File: rules/linux/persistence_rc_script_creation.toml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+type = "eql"
+
+query = '''
+sequence by user.id, host.id with maxspan=15s
+[file where event.type == "creation" and
+file.path : "/etc/rc.local"]
+[process where event.type == "start" and
+process.name == "chmod" and
+process.args : "+x"
+process.args : "/etc/rc.local"]
```
Comment (DefSecSentinel) [inline rules/linux/persistence_rc_script_creation.toml:68]:
```suggestion
sequence by user.id, host.id with maxspan=15s
[file where event.type == "creation" and
file.path : "/etc/rc.local"]
[process where event.type == "start" and
process.name == "chmod" and
process.args : "+x" and
process.args : "/etc/rc.local"]
```

**Review by DefSecSentinel — APPROVED** [review]
LGTM

File: rules/linux/persistence_rc_script_creation.toml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+type = "eql"
+
+query = '''
+sequence by user.id, host.id with maxspan=15s
+[file where event.type == "creation" and
+file.path : "/etc/rc.local"]
+[process where event.type == "start" and
+process.name == "chmod" and
+process.args : "+x" and
+process.args : "/etc/rc.local"]
```
Comment (brokensound77) [inline rules/linux/persistence_rc_script_creation.toml:68]: Can be more precise with case sensitivity on Linux. Also, some spacing makes it slightly more readable and less error prone

```suggestion
sequence by user.id, host.id with maxspan=15s
[file where event.type == "creation" and
   file.path == "/etc/rc.local"]
[process where event.type == "start" and
   process.name == "chmod" and
   process.args == "+x" and process.args == "/etc/rc.local"]
```

**Review by imays11 — APPROVED** [review]
Looks good, nice work 👍🏾

**Review by terrancedejesus — APPROVED** [review]
Looks good! Peer review with Ruben. The sequence of changing the properties of the file to be an executable should lower false-positive (FP) rates.

**Pull request merged**